{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import replaceReadsUtils\n",
    "import pysam\n",
    "import re\n",
    "import pandas\n",
    "from io import StringIO"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# set up some demo reads\n",
    "\n",
    "```\n",
    "    10\n",
    " a: AAAAAAAAAAAAAAAAAAA\n",
    " b:      TTTTTTTTTTTTTTTTTTTT\n",
    " m: AAAAAAAAAA-----AAAAACCCCC\n",
    " m2:       AAA---CCC\n",
    " ```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "genome = \"G\"*100\n",
    "genome_start = 3\n",
    "\n",
    "a = pysam.AlignedSegment()\n",
    "qseq = \"A\"*20\n",
    "a.query_name = \"read1\"\n",
    "a.query_sequence=qseq\n",
    "a.flag = 0\n",
    "a.reference_id = 0\n",
    "a.reference_start = 10\n",
    "a.mapping_quality = 20\n",
    "a.cigarstring = str(len(qseq)) + \"M\"\n",
    "a.query_qualities = pysam.qualitystring_to_array(\"<\"*len(qseq))\n",
    "a.tags = ((\"NM\", 1),(\"RG\", \"L1\"))\n",
    "\n",
    "b = pysam.AlignedSegment()\n",
    "qseq = \"T\"*20\n",
    "b.query_name = \"read2\"\n",
    "b.query_sequence=qseq\n",
    "b.flag = 0\n",
    "b.reference_id = 0\n",
    "b.reference_start = 15\n",
    "b.mapping_quality = 20\n",
    "b.cigarstring = str(len(qseq)) + \"M\"\n",
    "b.query_qualities = pysam.qualitystring_to_array(\"<\"*len(qseq))\n",
    "b.tags = ((\"NM\", 1),(\"RG\", \"L1\"))\n",
    "\n",
    "\n",
    "m = pysam.AlignedSegment()\n",
    "qseq = \"AAAAAAAAAAAAAAACCCCC\"\n",
    "m.query_name = \"mod\"\n",
    "m.query_sequence=qseq\n",
    "m.flag = 0\n",
    "m.reference_id = 0\n",
    "m.reference_start = 10\n",
    "m.mapping_quality = 20\n",
    "m.cigarstring = \"10M5D10M\"\n",
    "m.query_qualities = pysam.qualitystring_to_array(\"<\"*len(qseq))\n",
    "m.tags = ((\"NM\", 1),(\"RG\", \"L1\"))\n",
    "\n",
    "m2 = pysam.AlignedSegment()\n",
    "qseq = \"AAACCC\"\n",
    "m2.query_name = \"mod2\"\n",
    "m2.query_sequence=qseq\n",
    "m2.flag = 0\n",
    "m2.reference_id = 0\n",
    "m2.reference_start = 18\n",
    "m2.mapping_quality = 20\n",
    "m2.cigarstring = \"3M3D3M\"\n",
    "m2.query_qualities = pysam.qualitystring_to_array(\"<\"*len(qseq))\n",
    "m2.tags = ((\"NM\", 1),(\"RG\", \"L1\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Demo inserting the edit in m2 into read b\n",
    "\n",
    "Read b has 20 T's. Read m2 has 3 A's, a 3bp deletion, and 3 C's. We will insert this into read b."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "read b: read2\t0\t0\t15\t20\t20M\t-1\t-1\t20\tTTTTTTTTTTTTTTTTTTTT\tarray('B', [27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27])\t[('NM', 1), ('RG', 'L1')]\n",
      "read m2: mod2\t0\t0\t18\t20\t3M3D3M\t-1\t-1\t6\tAAACCC\tarray('B', [27, 27, 27, 27, 27, 27])\t[('NM', 1), ('RG', 'L1')]\n",
      "new:read2\t0\t0\t15\t20\t6M3D14M\t-1\t-1\t20\tTTTAAACCCTTTTTTTTGGG\tarray('B', [27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27, 27])\t[('NM', 1), ('RG', 'L1')]\n"
     ]
    }
   ],
   "source": [
    "new = replaceReadsUtils.replaceRead(b,m2,genome,genome_start)\n",
    "\n",
    "print(\"read b: \" + str(b))\n",
    "print(\"read m2: \" + str(m2))\n",
    "print(\"new:\" + str(new))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note that the new read is the same seqencing length as the original read b, but that it now includes a 3bp deletion as shown in the cigar sequence."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Demo creating simulated datasets at the EMX1 locus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "! if [ ! -f data/chr2.fa ]; then gunzip data/chr2.fa.gz; fi\n",
    "!mkdir -p out"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Namespace(alteredBam='data/ampSeq.bam', downsampleNumber=100, onlyIncludeAlteredWithIndel=True, outfile='out/demo.bam', qualAdd=0, reference='data/chr2.fa', swapChr='chr2', swapFreq=0.5, swapLoc=72933870, unalteredBam='data/unaltered.bam', unalteredNamesortedBam='data/unaltered.nameSort.bam')\n",
      "fetching chr2:72933870\n",
      "read 274 reads, kept 273 reads at target\n",
      "making changes in 50/273 sites\n",
      "sorting...\n",
      "sorting control...\n",
      "Finished\n",
      "read 16000 reads\n",
      "printed 5522 reads not at the cut site (downsample pct was 0.36496350364963503)\n",
      "printed 50 reads at cut site without modification\n",
      "printed 50 reads at cut site with modification\n",
      "printed 5622 reads to the treatment bam\n",
      "printed 6016 reads to the control bam\n",
      "\n"
     ]
    }
   ],
   "source": [
    "!python replaceReads.py --downsampleNumber 100 --swapFreq 0.5 \\\n",
    "    --qualAdd 0 --swapChr chr2 --swapLoc 72933870 --reference data/chr2.fa \\\n",
    "    --unalteredBam data/unaltered.bam --unalteredNamesortedBam data/unaltered.nameSort.bam \\\n",
    "    --alteredBam data/ampSeq.bam \\\n",
    "    --outfile out/demo.bam  \\\n",
    "    --onlyIncludeAlteredWithIndel"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Run Pindel on simulated dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "#the -e flag doesn't exist in all echo versions\n",
    "!echo \"out/demo.bam.ctl.bam\t550\tNORMAL\" > out/demo.pindelConfig\n",
    "!echo \"out/demo.bam\t550\tTUMOR\" >> out/demo.pindelConfig"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Initializing parameters...\n",
      "Pindel version 0.2.5b8, 20151210.\n",
      "Loading reference genome ...\n",
      "Loading reference genome done.\n",
      "Initializing parameters done.\n",
      "SearchRegion::SearchRegion\n",
      "Processing region: chr2\t1\t242193529\n",
      "Chromosome Size: 242193530\n",
      "NumBoxes: 60006\tBoxSize: 8079\n",
      "\n",
      "Looking at chromosome chr2 bases 1 to 5000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 5000001 to 10000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 10000001 to 15000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 15000001 to 20000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 20000001 to 25000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 25000001 to 30000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 30000001 to 35000001 of the bed region: chromosome chr2:1-242193529 \n",
      "out/demo.bam.ctl.bam RP 2\n",
      "out/demo.bam RP 7\n",
      "Discovery RP: 7\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 7\n",
      "sorting read-pair\n",
      "sorting read-pair finished.\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t2, + 1 - 1\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t7, + 4 - 3\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 35000001 to 40000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 40000001 to 45000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 45000001 to 50000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 50000001 to 55000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 55000001 to 60000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 60000001 to 65000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 65000001 to 70000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 70000001 to 75000001 of the bed region: chromosome chr2:1-242193529 \n",
      "out/demo.bam.ctl.bam RP 2\n",
      "out/demo.bam RP 7\n",
      "Discovery RP: 7\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 7\n",
      "sorting read-pair\n",
      "sorting read-pair finished.\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 756\n",
      "Number of problematic reads in current window:            \t1912, + 1461 - 451\n",
      "Number of split-reads where the close end could be mapped:\t756, + 586 - 170\n",
      "Percentage of problematic reads with close end mapped:    \t+ 40.11% - 37.69%\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t756\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 1586\n",
      "Number of problematic reads in current window:            \t3862, + 2935 - 927\n",
      "Number of split-reads where the close end could be mapped:\t1586, + 1233 - 353\n",
      "Percentage of problematic reads with close end mapped:    \t+ 42.01% - 38.08%\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t1586\n",
      "\n",
      "The number of one end mapped read: 1586\n",
      "There are 10988 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "There are 1586 split-reads for this chromosome region.\n",
      "\n",
      "There are 0 split-reads mapped by aligner.\n",
      "search far ends\n",
      "Far end searching completed for this window.\n",
      "update FarFragName\n",
      "update FarFragName done\n",
      "Searching and reporting variations\n",
      "Reads already used: 0\n",
      "Far ends already mapped 1031\n",
      "Checksum of far ends: 298936971\n",
      "Searching some type of variant, replace this with the correct name in child class ... \n",
      "Total: 85\t+64\t-21\n",
      "Sorting and outputing deletions ...\n",
      "Deletions: 27\n",
      "\n",
      "Searching deletion-insertions ... \n",
      "Total: 10\t+7\t-3\n",
      "Sorting and outputing deletions with non-template sequences ...\n",
      "Added: Sorting and outputing deletions with non-template sequences ...\n",
      "deletions with non-template sequences: 3\n",
      "\n",
      "Searching tandem duplication events ... \n",
      "Total: 0\t+0\t-0\n",
      "Sorting and outputing tandem duplications ...\n",
      "Tandem duplications: 0\n",
      "\n",
      "Searching tandem duplication events with non-template sequence ... \n",
      "Total: 4\t+3\t-1\n",
      "Sorting and outputing tandem duplications with non-template sequence ...\n",
      "Tandem duplications with non-template sequence (TD_NT): 0\n",
      "\n",
      "Searching inversions ... \n",
      "Total: 5\t+4\t-1\n",
      "Sorting and outputing Inversions ...\n",
      "Inversions (INV): 1\n",
      "\n",
      "Searching inversions with non-template sequence ... \n",
      "Total: 25\t+24\t-1\n",
      "Sorting and outputing Inversions with non-template sequence ...\n",
      "Inversions with non-template sequence (INV_NT): 2\n",
      "\n",
      "Reads already used: 129\n",
      "Far ends already mapped 1031\n",
      "Checksum of far ends: 298936971\n",
      "Searching some type of variant, replace this with the correct name in child class ... \n",
      "Total: 315\t+232\t-83\n",
      "Sorting and outputing short insertions ...\n",
      "Short insertions: 4\n",
      "\n",
      "Total: 1586;\tClose_end_found 1586;\tFar_end_found 1031;\tUsed\t444.\n",
      "\n",
      "For LI and BP: 555\n",
      "\n",
      "There are 0 split-reads saved for the next cycle.\n",
      "\n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 75000001 to 80000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 80000001 to 85000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 85000001 to 90000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 90000001 to 95000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 95000001 to 100000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 100000001 to 105000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 105000001 to 110000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 110000001 to 115000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 115000001 to 120000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 120000001 to 125000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 125000001 to 130000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 130000001 to 135000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 135000001 to 140000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 140000001 to 145000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 145000001 to 150000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 150000001 to 155000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 155000001 to 160000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 160000001 to 165000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 165000001 to 170000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 170000001 to 175000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 175000001 to 180000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 180000001 to 185000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 185000001 to 190000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 190000001 to 195000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 195000001 to 200000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 200000001 to 205000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 205000001 to 210000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 210000001 to 215000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 215000001 to 220000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 220000001 to 225000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 225000001 to 230000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 230000001 to 235000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 235000001 to 240000001 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 5000001 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "\n",
      "Looking at chromosome chr2 bases 240000001 to 242193529 of the bed region: chromosome chr2:1-242193529 \n",
      "No discordant RP reads in Bamfile out/demo.bam.ctl.bam\n",
      "No discordant RP reads in Bamfile out/demo.bam\n",
      "Discovery RP: 0\n",
      "sorting RP complete.\n",
      "Reads_RP.size(): 0\n",
      "Modify RP complete.\n",
      "adding BD from RP.\n",
      "modify and summarize interchr RP.\n",
      "adding BD from interChr RP.\n",
      "summarize BP as BD complete. Now start sorting BD...\n",
      "sorting BD... done.\n",
      "external BD events: 0 Added BreakDancer-like events: 0\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam.ctl.bam\n",
      "BAM file index\t0\n",
      "Bam file name\tout/demo.bam.ctl.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "\n",
      "Insertsize in config: 550\n",
      "The number of one end mapped read: 0\n",
      "Number of problematic reads in current window:            \t0, + 0 - 0\n",
      "Number of split-reads where the close end could be mapped:\t0, + 0 - 0\n",
      "Percentage of problematic reads with close end mapped:    \t+ 0.00% - 0.00%\n",
      "No currentState.Reads for chr2 found in out/demo.bam\n",
      "BAM file index\t1\n",
      "Bam file name\tout/demo.bam\n",
      "Number of split-reads so far\t0\n",
      "\n",
      "The number of one end mapped read: 0\n",
      "There are 0 reads supporting the reference allele.\n",
      "There are 2 samples.\n",
      "SampleName2Index done\n",
      "declaring g_RefCoverageRegion for 2 samples and 2193531 positions.\n",
      "no reads \n",
      "InterChromosome_SR.size(): 0\n",
      "PreviousChrName: \n",
      "Initializing pindel 5 seconds.\n",
      "Loading chromosomes 1 seconds.\n",
      "Reading in reads + matching close ends 10 seconds.\n",
      "Searching far ends 1 seconds.\n",
      "Searching and reporting variations 0 seconds.\n"
     ]
    }
   ],
   "source": [
    "!pindel -f data/chr2.fa -i out/demo.pindelConfig -o out/demo.pindel"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Inserted mutations:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BP\tINDEL\tLOC\tCOUNT\r\n",
      "21\tD\t72933855\t3\r\n",
      "21\tD\t72933861\t1\r\n",
      "1\tD\t72933918\t1\r\n",
      "2\tD\t72933924\t1\r\n",
      "1\tD\t72933867\t6\r\n",
      "46\tD\t72933824\t1\r\n",
      "1\tI\t72933867\t20\r\n",
      "3\tD\t72933862\t4\r\n",
      "2\tD\t72933730\t1\r\n",
      "2\tI\t72933739\t1\r\n",
      "2\tD\t72933868\t1\r\n",
      "6\tD\t72933862\t3\r\n",
      "30\tD\t72933853\t1\r\n",
      "2\tI\t72933868\t1\r\n",
      "9\tD\t72933859\t1\r\n",
      "1\tI\t72933868\t1\r\n",
      "28\tD\t72933849\t1\r\n",
      "1\tD\t72933864\t1\r\n",
      "7\tD\t72933866\t1\r\n",
      "21\tD\t72933854\t1\r\n",
      "1\tD\t72933766\t1\r\n",
      "29\tD\t72933847\t1\r\n"
     ]
    }
   ],
   "source": [
    "!more out/demo.bam.mutations.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Deletions found by Pindel:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "####################################################################################################\r\n",
      "0\tD 2\tNT 0 \"\"\tChrID chr2\tBP 72928818\t72928821\tBP_range 72928818\t72928822\tSupports 1\t1\t+ 0\t0\t- 1\t1\tS1 2\tSUM_MS 60\t2\tNumSupSamples 1\t1\tNORMAL 27 27 0 0 0 0\tTUMOR 33 33 0 0 1 1\r\n",
      "GGGTACAGAAACGTGGCCTGTGGCTTTGAAGATTTCTTACTGAGTGATGAAGGCTAAGTGCAAAGCTTGCACATTTGTGAAACATGCACAGGAAGAATGACTAGGGTCCCCTTTGAGGTCACAGCTGTGGGCTGAGGGGTGTCAGGATagAAACGCTTGAGAAGATGGCTCCAGGAGGCCTCAGACCTGGAAGACTTGGGGGGATGCCTAGGATCCTAGGTTGGAGGGAAAGAAGGGCAGGGTTTGAGGCAGGCAGATGAAGATAGAGCCACCATCTTGGAGCCCAAGGGCAGGGA\r\n",
      "                                                                                                                                   CTGAGGGGTGTCAGGAT  AAACGCTTGAGAAGATGGCTCCAGGAGGCCTCAGACCTGGAAGACTTGGGGGGATGCCTAGGATCCTAGGTTGGAGGGAAAGAAGGGCAGGGTTTGAGGCAGGCAGATGAAGATAGAGCCACCATCTTGGA                 \t-\t72929308\t60\tTUMOR\t@HWI-D00360:8:H88U0ADXX:2:2116:16314:69264/1\r\n",
      "####################################################################################################\r\n",
      "1\tD 1\tNT 0 \"\"\tChrID chr2\tBP 72929595\t72929597\tBP_range 72929595\t72929599\tSupports 1\t1\t+ 0\t0\t- 1\t1\tS1 2\tSUM_MS 60\t2\tNumSupSamples 1\t1\tNORMAL 80 80 0 0 1 1\tTUMOR 92 92 0 0 0 0\r\n",
      "AGCATGGCATTGATTGGGCTTGACCTTGAGAGTGGGAACAGCCCACATGATCAAATGATAGGAAATGGGTCCCCAGGGAAGAGAGAAAGCTGAGGGGTGACTTGACTGATATCTCCAGGCTCCTGTCCATTCTGGGATATTTGAAATCtTTGAAGACAGGACACGTATTCACCTGAAAGTGTGCAGGGGCCTGGAGCCACAGACTTTTCCATTTGCAGGAGTGGTGAAAAAGAAGGGATCCAGACCTCGTTCATTCAGAATAGCAGATTATTCCAAAGTAAAGTCTGATTAGTTCT\r\n",
      "                                                                      CCCCAGGGAAGAGAGAAAGCTGAGGGGTGACTTGACTGATATCTCCAGGCTCCTGTCCATTCTGGGATATTTGAAATC TTGAAGACAGGACACGTATTCACCTGAAAGTGTGCAGGGGCCTGGAGCCACAGACTTTTCCATTTGCAGG                                                                              \t-\t72930093\t60\tNORMAL\t@HISEQ1:15:H9258ADXX:1:2213:20251:67673/1\r\n",
      "####################################################################################################\r\n",
      "2\tD 1\tNT 0 \"\"\tChrID chr2\tBP 72930391\t72930393\tBP_range 72930391\t72930394\tSupports 1\t1\t+ 1\t1\t- 0\t0\tS1 2\tSUM_MS 60\t2\tNumSupSamples 1\t1\tNORMAL 96 96 0 0 0 0\tTUMOR 90 90 1 1 0 0\r\n"
     ]
    }
   ],
   "source": [
    "!head out/demo.pindel_D"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "from collections import defaultdict\n",
    "\n",
    "resDict = {}\n",
    "variantsRecovered = defaultdict(int) #variant -> count recovered e.g. 1D -> 5\n",
    "variantsSimulated = defaultdict(int) # variant -> count simulated e.g. 1D -> 10 (1D was simulated 10 times, and recovered 5 times)\n",
    "variantsFalseRecovered = {} # variant -> count not simulated (FDR)\n",
    "\n",
    "\n",
    "cutsIn = \"out/demo.bam.info\"\n",
    "cutsFile = open(cutsIn,\"r\")\n",
    "readsNotAtSite = \"NA\"\n",
    "readsAtSiteNoMod = \"NA\"\n",
    "readsAtSiteMod = \"NA\"\n",
    "readReads = \"NA\"\n",
    "for line in cutsFile:\n",
    "        m = re.search(\"printed (\\d+) reads at cut site without modification\",line)\n",
    "        if(m):\n",
    "                readsAtSiteNoMod = m.group(1)\n",
    "        m = re.search(\"printed (\\d+) reads at cut site with modification\",line)\n",
    "        if(m):\n",
    "                readsAtSiteMod = m.group(1)\n",
    "        m = re.search(\"printed (\\d+) reads not at the cut site\",line)\n",
    "        if(m):\n",
    "                readsNotAtSite = m.group(1)\n",
    "        m = re.search(\"read (\\d+) reads\",line)\n",
    "        if(m):\n",
    "                readReads = m.group(1)\n",
    "cutsFile.close()\n",
    "\n",
    "mutsIn = \"out/demo.bam.mutations.txt\"\n",
    "mutsFile = open(mutsIn,\"r\")\n",
    "theseSimulatedMuts = {}\n",
    "headLine = mutsFile.readline()\n",
    "#BP INDEL LOC COUNT\n",
    "for line in mutsFile:\n",
    "        line.strip()\n",
    "        lineEls = line.split()\n",
    "        indelKey = lineEls[0] + \" \" + lineEls[1] + \" \" + lineEls[2]\n",
    "        theseSimulatedMuts[indelKey] = int(lineEls[3])\n",
    "mutsFile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "theseFoundMuts = {}\n",
    "\n",
    "delCounts = 0\n",
    "insCounts = 0\n",
    "pindelSupport = 0\n",
    "pindelUniqueSupport = 0\n",
    "\n",
    "pindelFileNames = [\"out/demo.pindel_D\",\n",
    "                \"out/demo.pindel_SI\",\n",
    "                \"out/demo.pindel_LI\",\n",
    "                ]\n",
    "for pindelFileName in pindelFileNames:\n",
    "        pindelFile = open(pindelFileName,\"r\")\n",
    "        for line in pindelFile:\n",
    "                lineEls = line.split()\n",
    "                if len(lineEls) > 20:\n",
    "                        if lineEls[25] != \"SUM_MS\":\n",
    "                                raise Exception(\"Can't read line \" + line + \" (got lineEls[25] = \" + lineEls[25] + \")\")\n",
    "\n",
    "                        #BP INDEL LOC => COUNT\n",
    "                        thisKey = lineEls[2] + \" \" + lineEls[1] + \" \" + lineEls[9]\n",
    "                        theseFoundMuts[thisKey] = int(lineEls[15]) #pindel Support\n",
    "\n",
    "        pindelFile.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "thisSimulatedCount = 0\n",
    "thisSimulatedRecoveredCount = 0\n",
    "for simulatedKey in theseSimulatedMuts:\n",
    "        thisSimulatedCount += theseSimulatedMuts[simulatedKey]\n",
    "        thisRecoveredCount = 0\n",
    "        if simulatedKey in theseFoundMuts:\n",
    "                thisRecoveredCount = theseFoundMuts[simulatedKey]\n",
    "                thisSimulatedRecoveredCount += thisRecoveredCount\n",
    "\n",
    "        (simLen,simType,simLoc) = simulatedKey.split()\n",
    "        varKey = simLen + \"\\t\" + simType\n",
    "\n",
    "        variantsRecovered[varKey] += thisRecoveredCount  #variant -> count recovered e.g. 1D -> 5\n",
    "        variantsSimulated[varKey] += theseSimulatedMuts[simulatedKey] # variant -> count simulated e.g. 1D -> 10 (1D was simulated 10 times, and recovered 5 times)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "swapLoc = 72933870 #site of CRISPR mutation\n",
    "tolerance = [10,50,100] #how far away from site we should look for false positives\n",
    "\n",
    "thisFoundCount = 0\n",
    "thisFoundSimulatedCount = 0\n",
    "for foundKey in theseFoundMuts:\n",
    "        thisFoundCount += theseFoundMuts[foundKey]\n",
    "        if foundKey in theseSimulatedMuts:\n",
    "                thisFoundSimulatedCount += theseSimulatedMuts[foundKey]\n",
    "        else:\n",
    "                (foundLen,foundType,foundLoc) = foundKey.split()\n",
    "                #make sure this was within 5bp from cut site.. don't want to include too many false positives\n",
    "                varKey = foundLen + \"\\t\" + foundType\n",
    "                for t in tolerance:\n",
    "                        if abs(int(foundLoc) - swapLoc) <= t:\n",
    "                                if varKey not in variantsFalseRecovered:\n",
    "                                        variantsFalseRecovered[varKey] = defaultdict(int)\n",
    "                                variantsFalseRecovered[varKey][t] += theseFoundMuts[foundKey]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style>\n",
       "    .dataframe thead tr:only-child th {\n",
       "        text-align: right;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: left;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Length</th>\n",
       "      <th>Type</th>\n",
       "      <th>Simulated</th>\n",
       "      <th>Recovered</th>\n",
       "      <th>FDR10</th>\n",
       "      <th>FDR50</th>\n",
       "      <th>FDR100</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>D</td>\n",
       "      <td>9</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>I</td>\n",
       "      <td>21</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>D</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2</td>\n",
       "      <td>I</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>3</td>\n",
       "      <td>D</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>6</td>\n",
       "      <td>D</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>7</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>9</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>21</td>\n",
       "      <td>D</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>28</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>29</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>30</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>31</td>\n",
       "      <td>D</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>46</td>\n",
       "      <td>D</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Length Type  Simulated  Recovered  FDR10  FDR50  FDR100\n",
       "0        1    D          9          7      0      0       0\n",
       "1        1    I         21         19      0      0       0\n",
       "2        2    D          3          0      0      0       0\n",
       "3        2    I          2          0      0      0       0\n",
       "7        3    D          4          4      0      0       0\n",
       "11       6    D          3          3      0      0       0\n",
       "12       7    D          1          1      0      0       0\n",
       "13       9    D          1          1      0      0       0\n",
       "4       21    D          5          3      0      0       0\n",
       "5       28    D          1          1      0      0       0\n",
       "6       29    D          1          1      0      0       0\n",
       "8       30    D          1          0      0      0       0\n",
       "9       31    D          0          0      0      1       1\n",
       "10      46    D          1          1      0      0       0"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "outStr = \"Length\\tType\\tSimulated\\tRecovered\"\n",
    "for t in tolerance:\n",
    "    outStr += \"\\tFDR\"+str(t)\n",
    "outStr += \"\\n\"\n",
    "for mutKey in sorted(set(variantsSimulated.keys()).union(variantsFalseRecovered.keys())):\n",
    "        outStr += mutKey+\"\\t\"+str(variantsSimulated[mutKey]) + \"\\t\" + str(variantsRecovered[mutKey])\n",
    "        if mutKey in variantsFalseRecovered:\n",
    "                for t in tolerance:\n",
    "                        outStr += \"\\t\" + str(variantsFalseRecovered[mutKey][t])\n",
    "        else:\n",
    "                for t in tolerance:\n",
    "                        outStr += \"\\t0\"\n",
    "        outStr += \"\\n\"\n",
    "\n",
    "#print(outStr)\n",
    "df = pandas.read_table(StringIO(outStr))\n",
    "df.sort_values('Length')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.collections.PathCollection at 0x2b6683820a90>"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAX4AAAD9CAYAAAC7iRw+AAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAExNJREFUeJzt3X+IH/d95/Hn62QZhGOQc9molmxdEjAKvtDI7eKEq+9w\nmjiyRajVo/RsjtbpBZSUpCTQ053dQFsOAgGR9Og5xKgXYwdSpymRFUPV6JwQcAJJ6pWtWvIP1apx\nsFaKpNTIdi4LtpT3/bGzvq/X39V+9Z3v6rvaeT7gy3fmM5+Zee/w5aWvPjPfmVQVkqTu+FfjLkCS\ndGEZ/JLUMQa/JHWMwS9JHWPwS1LHGPyS1DGLBn+Sq5N8L8lTSZ5M8umm/a1JHk7ybPN+xQLr35zk\ncJIjSe4c9R8gSTo/Wew6/iRXAldW1WNJLgf2A9uAjwIvVtXnm0C/oqr++7x1VwH/BNwEHAUeBW6v\nqqdG/pdIkgay6Df+qjpeVY81068ATwMbgFuB+5tu9zP7j8F81wNHquq5qnoV+HqzniRpTM5rjD/J\nO4DrgB8D66rqeLPop8C6PqtsAF7omT/atEmSxuSSQTsmeQvwTeAzVfVykteXVVUlaXXvhyTbge0A\nl1122a+/+93vbrM5SeqU/fv3/6yqJgbpO1DwJ1nNbOh/rap2N80nklxZVceb8wAn+6w6DVzdM39V\n0/YmVbUL2AUwOTlZU1NTg5QmSQKS/GTQvoNc1RPgK8DTVfXFnkUPAXc003cA3+qz+qPANUnemeRS\n4LZmPUnSmAwyxv8bwO8Bv5nkQPPaCnweuCnJs8CHmnmSrE+yF6CqzgCfAvYxe1L4G1X15BL8HZKk\nAS061FNVPwCywOIP9ul/DNjaM78X2DtsgZKk0fKXu5LUMQa/JHXMwJdzSpKWxp7Hp9m57zDHTs+w\nfu0admzZxLbrlu4nTwa/JI3RnsenuWv3QWZeOwvA9OkZ7tp9EGDJwt+hHkkao537Dr8e+nNmXjvL\nzn2Hl2yfBr8kjdGx0zPn1T4KBr8kjdH6tWvOq30UDH5JGqMdWzaxZvWqN7StWb2KHVs2Ldk+Pbkr\nSWM0dwLXq3okqUO2XbdhSYN+Pod6JKljDH5J6hiDX5I6xuCXpI4x+CWpYwx+SeoYg1+SOmbR6/iT\n3At8BDhZVe9p2v4GmPtZ2VrgdFVt7rPu88ArwFngTFVNjqhuSdKQBvkB133A3cBX5xqq6j/NTSf5\nAvDSOdb/QFX9bNgCJUmjNcgzdx9J8o5+y5IE+F3gN0dbliRpqbQd4//3wImqenaB5QV8J8n+JNtb\n7kuSNAJt79VzO/DAOZbfUFXTSd4OPJzkmap6pF/H5h+G7QAbN25sWZYkaSFDf+NPcgnwH4G/WahP\nVU037yeBB4Hrz9F3V1VNVtXkxMTEsGVJkhbRZqjnQ8AzVXW038IklyW5fG4a+DBwqMX+JEkjsGjw\nJ3kA+CGwKcnRJB9rFt3GvGGeJOuT7G1m1wE/SPKPwD8Af1dV3x5d6ZKkYQxyVc/tC7R/tE/bMWBr\nM/0c8N6W9UmSRsxf7kpSxxj8ktQxBr8kdYzBL0kdY/BLUscY/JLUMQa/JHWMwS9JHWPwS1LHGPyS\n1DEGvyR1jMEvSR1j8EtSxxj8ktQxBr8kdYzBL0kdY/BLUscM8ujFe5OcTHKop+3Pk0wnOdC8ti6w\n7s1JDic5kuTOURYuSRrOIN/47wNu7tP+F1W1uXntnb8wySrgS8AtwLXA7UmubVOsJKm9RYO/qh4B\nXhxi29cDR6rquap6Ffg6cOsQ25EkjVCbMf4/SvJEMxR0RZ/lG4AXeuaPNm2SpDEaNvi/DLwL2Awc\nB77QtpAk25NMJZk6depU281JkhYwVPBX1YmqOltVvwT+itlhnfmmgat75q9q2hba5q6qmqyqyYmJ\niWHKkiQNYKjgT3Jlz+xvA4f6dHsUuCbJO5NcCtwGPDTM/iRJo3PJYh2SPADcCLwtyVHgz4Abk2wG\nCnge+HjTdz3wv6tqa1WdSfIpYB+wCri3qp5ckr9CkjSwVNW4a3iTycnJmpqaGncZknTRSLK/qiYH\n6esvdyWpYwx+SeoYg1+SOsbgl6SOMfglqWMMfknqGINfkjrG4JekjjH4JaljDH5J6hiDX5I6xuCX\npI4x+CWpYwx+SeoYg1+SOsbgl6SOMfglqWMWDf4k9yY5meRQT9vOJM8keSLJg0nWLrDu80kOJjmQ\nxEdqSdIyMMg3/vuAm+e1PQy8p6p+Ffgn4K5zrP+Bqto86CPBJElLa9Hgr6pHgBfntf2fqjrTzP4I\nuGoJapMkLYFRjPH/F+DvF1hWwHeS7E+yfQT7kiS1dEmblZN8FjgDfG2BLjdU1XSStwMPJ3mm+R9E\nv21tB7YDbNy4sU1ZkqRzGPobf5KPAh8B/nNVVb8+VTXdvJ8EHgSuX2h7VbWrqiaranJiYmLYsiRJ\nixgq+JPcDPw34Leq6hcL9LksyeVz08CHgUP9+kqSLpxBLud8APghsCnJ0SQfA+4GLmd2+OZAknua\nvuuT7G1WXQf8IMk/Av8A/F1VfXtJ/gpJ0sAWHeOvqtv7NH9lgb7HgK3N9HPAe1tVJ0kaOX+5K0kd\nY/BLUscY/JLUMQa/JHWMwS9JHWPwS1LHGPyS1DEGvyR1jMEvSR1j8EtSxxj8ktQxBr8kdYzBL0kd\nY/BLUscY/JLUMQa/JHWMwS9JHTPIoxfvTXIyyaGetrcmeTjJs837FQuse3OSw0mOJLlzlIVLkoYz\nyDf++4Cb57XdCXy3qq4BvtvMv0GSVcCXgFuAa4Hbk1zbqlpJUmuLBn9VPQK8OK/5VuD+Zvp+YFuf\nVa8HjlTVc1X1KvD1Zj1J0hgNO8a/rqqON9M/Bdb16bMBeKFn/mjTJkkao9Ynd6uqgGq7nSTbk0wl\nmTp16lTbzUmSFjBs8J9IciVA836yT59p4Oqe+auatr6qaldVTVbV5MTExJBlSZIWM2zwPwTc0Uzf\nAXyrT59HgWuSvDPJpcBtzXqSpDEa5HLOB4AfApuSHE3yMeDzwE1JngU+1MyTZH2SvQBVdQb4FLAP\neBr4RlU9uTR/hiRpUJcs1qGqbl9g0Qf79D0GbO2Z3wvsHbo6SdLI+ctdSeoYg1+SOsbgl6SOMfgl\nqWMMfknqGINfkjrG4JekjjH4JaljDH5J6hiDX5I6xuCXpI4x+CWpYwx+SeoYg1+SOsbgl6SOMfgl\nqWMMfknqmKGDP8mmJAd6Xi8n+cy8Pjcmeamnz5+2L1mS1Maij15cSFUdBjYDJFkFTAMP9un6/ar6\nyLD7kSSN1qiGej4I/HNV/WRE25MkLZFRBf9twAMLLPt3SZ5I8vdJ/u2I9idJGlLr4E9yKfBbwN/2\nWfwYsLGqfhX4X8Cec2xne5KpJFOnTp1qW5YkaQGj+MZ/C/BYVZ2Yv6CqXq6qnzfTe4HVSd7WbyNV\ntauqJqtqcmJiYgRlSZL6Gfrkbo/bWWCYJ8mvACeqqpJcz+w/NP8ygn1Kfe15fJqd+w5z7PQM69eu\nYceWTWy7bsO4y5KWlVbBn+Qy4Cbg4z1tnwCoqnuA3wH+MMkZYAa4raqqzT6lhex5fJq7dh9k5rWz\nAEyfnuGu3QcBDH+pR6vgr6r/C/zreW339EzfDdzdZh/SoHbuO/x66M+Zee0sO/cdNvilHv5yVyvG\nsdMz59UudZXBrxVj/do159UudZXBrxVjx5ZNrFm96g1ta1avYseWTWOqSFqeRnFVj7QszI3je1WP\ndG4Gv1aUbddtMOilRTjUI0kdY/BLUscY/JLUMQa/JHWMwS9JHWPwS1LHGPyS1DEGvyR1jMEvSR1j\n8EtSxxj8ktQxBr8kdUyr4E/yfJKDSQ4kmeqzPEn+MsmRJE8k+bU2+5MktTeKu3N+oKp+tsCyW4Br\nmtf7gC8375KkMVnqoZ5bga/WrB8Ba5NcucT7lCSdQ9vgL+A7SfYn2d5n+QbghZ75o02bJGlM2g71\n3FBV00neDjyc5JmqemSYDTX/cGwH2LhxY8uyJEkLafWNv6qmm/eTwIPA9fO6TANX98xf1bT129au\nqpqsqsmJiYk2ZUmSzmHo4E9yWZLL56aBDwOH5nV7CPj95uqe9wMvVdXxoauVJLXWZqhnHfBgkrnt\n/HVVfTvJJwCq6h5gL7AVOAL8AviDduVKktoaOvir6jngvX3a7+mZLuCTw+5DkjR6/nJXkjrG4Jek\njhnFL3d1Ae15fJqd+w5z7PQM69euYceWTWy7zp9GSBqcwX8R2fP4NHftPsjMa2cBmD49w127DwIY\n/pIG5lDPRWTnvsOvh/6cmdfOsnPf4TFVJOliZPBfRI6dnjmvdknqx+C/iKxfu+a82iWpH4P/IrJj\nyybWrF71hrY1q1exY8umMVUk6WLkyd2LyNwJXK/qkdSGwX+R2XbdBoNeUisO9UhSxxj8ktQxBr8k\ndYxj/GMwjtsueKsHSXMM/gtsHLdd8FYPkno51HOBjeO2C97qQVIvg/8CG8dtF7zVg6RebZ65e3WS\n7yV5KsmTST7dp8+NSV5KcqB5/Wm7ci9+47jtgrd6kNSrzTf+M8AfV9W1wPuBTya5tk+/71fV5ub1\nP1rsb0UYx20XvNWDpF5tnrl7HDjeTL+S5GlgA/DUiGpbkcZx2wVv9SCpV2afh95yI8k7gEeA91TV\nyz3tNwK7gaPANPBfq+rJxbY3OTlZU1NTreuSpK5Isr+qJgfp2/pyziRvAb4JfKY39BuPARur6udJ\ntgJ7gGsW2M52YDvAxo0b25YlSVpAq6t6kqxmNvS/VlW75y+vqper6ufN9F5gdZK39dtWVe2qqsmq\nmpyYmGhTliTpHNpc1RPgK8DTVfXFBfr8StOPJNc3+/uXYfcpSWqvzVDPbwC/BxxMcqBp+xNgI0BV\n3QP8DvCHSc4AM8BtNYqTCpKkobW5qucHQBbpczdw97D7kCSNnr/claSOMfglqWMMfknqGINfkjrG\n4JekjjH4JaljDH5J6pjOP3rxfZ97mBOvvPr6/LrLL+XHn71pjBVJ0tLq9Df++aEPcOKVV3nf5x4e\nU0WStPQ6HfzzQ3+xdklaCTod/JLURQa/JHVMp4N/3eWXnle7JK0EnQ7+H3/2pjeFvFf1SFrpOn85\npyEvqWs6/Y1fkrrI4Jekjmn7sPWbkxxOciTJnX2WJ8lfNsufSPJrbfYnSWpv6DH+JKuALwE3AUeB\nR5M8VFVP9XS7Bbimeb0P+HLzPnJ7Hp9m577DHDs9w/q1a9ixZRPbrtuwZOuNa5+S1Fabk7vXA0eq\n6jmAJF8HbgV6g/9W4KvNA9Z/lGRtkiur6niL/b7JnsenuWv3QWZeOwvA9OkZ7tp9EOCcgTrseuPa\npySNQpuhng3ACz3zR5u28+3T2s59h18P0jkzr51l577DS7LeuPYpSaOwbE7uJtmeZCrJ1KlTp85r\n3WOnZ86rve1649qnJI1Cm+CfBq7umb+qaTvfPgBU1a6qmqyqyYmJifMqZP3aNefV3na9ce1Tkkah\nTfA/ClyT5J1JLgVuAx6a1+ch4Pebq3veD7w06vF9gB1bNrFm9ao3tK1ZvYodWzYtyXrj2qckjcLQ\nJ3er6kySTwH7gFXAvVX1ZJJPNMvvAfYCW4EjwC+AP2hf8pvNnRQ93ytlhl1vXPuUpFHI7AU3y8vk\n5GRNTU2NuwxJumgk2V9Vk4P0XTYndyVJF4bBL0kdY/BLUscY/JLUMQa/JHXMsryqJ8kp4CdDrv42\n4GcjLGel8fgszmN0bh6fxY3jGP2bqhro16/LMvjbSDI16CVNXeTxWZzH6Nw8Potb7sfIoR5J6hiD\nX5I6ZiUG/65xF7DMeXwW5zE6N4/P4pb1MVpxY/ySpHNbid/4JUnnsGKCf7EHvwuSPJ/kYJIDSTp/\nF7wk9yY5meRQT9tbkzyc5Nnm/Ypx1jhuCxyjP08y3XyODiTZOs4axynJ1Um+l+SpJE8m+XTTvqw/\nRysi+Hse/H4LcC1we5Jrx1vVsvWBqtq8nC81u4DuA26e13Yn8N2qugb4bjPfZffx5mME8BfN52hz\nVe29wDUtJ2eAP66qa4H3A59ssmdZf45WRPDT8+D3qnoVmHvwu7SgqnoEeHFe863A/c30/cC2C1rU\nMrPAMVKjqo5X1WPN9CvA08w+V3xZf45WSvBfkIe6rwAFfCfJ/iTbx13MMrWu5ylxPwXWjbOYZeyP\nkjzRDAUtq2GMcUnyDuA64Mcs88/RSgl+DeaGqtrM7JDYJ5P8h3EXtJzV7CVvXvb2Zl8G3gVsBo4D\nXxhvOeOX5C3AN4HPVNXLvcuW4+dopQT/wA9177Kqmm7eTwIPMjtEpjc6keRKgOb95JjrWXaq6kRV\nna2qXwJ/Rcc/R0lWMxv6X6uq3U3zsv4crZTgH+TB752W5LIkl89NAx8GDp17rU56CLijmb4D+NYY\na1mW5gKt8dt0+HOUJMBXgKer6os9i5b152jF/ICruaTsf/L/H/z+uTGXtKwkeRez3/IBLgH+uuvH\nKMkDwI3M3knxBPBnwB7gG8BGZu8Q+7tV1dmTmwscoxuZHeYp4Hng4z3j2Z2S5Abg+8BB4JdN858w\nO86/bD9HKyb4JUmDWSlDPZKkARn8ktQxBr8kdYzBL0kdY/BLUscY/JLUMQa/JHWMwS9JHfP/AIHQ\nix2r8wWRAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x2b66762c0908>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.scatter(df.Simulated,df.Recovered)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
